packages/list_bloc/test/cubits/list_cubit_test.dart

```diff
@@ -128,5 +128,137 @@ void main() {
         ),
       ],
     );
+
+    blocTest<ListCubit<Item, ItemType>, Data<List<Item>, ItemType>>(
+      'is able to add an item correctly',
+      build: () => cubit,
+      act: (cubit) async {
+        await cubit.load(ItemType.fruit);
+        cubit.add(Item(type: ItemType.vegetable, value: 'Carrot'));
+      },
+      expect: () => [
+        Data<List<Item>, ItemType>.loading(
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [Item(type: ItemType.fruit, value: 'Apple')],
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [
+            Item(type: ItemType.fruit, value: 'Apple'),
+            Item(type: ItemType.vegetable, value: 'Carrot'),
+          ],
+          filter: ItemType.fruit,
+        ),
+      ],
+    );
+
+    blocTest<ListCubit<Item, ItemType>, Data<List<Item>, ItemType>>(
+      'is able to remove an item by index correctly',
+      build: () => cubit,
+      act: (cubit) async {
+        await cubit.load(ItemType.fruit);
+        cubit.add(Item(type: ItemType.vegetable, value: 'Carrot'));
+        cubit.removeAt(1);
+      },
+      expect: () => [
+        Data<List<Item>, ItemType>.loading(
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [Item(type: ItemType.fruit, value: 'Apple')],
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [
+            Item(type: ItemType.fruit, value: 'Apple'),
+            Item(type: ItemType.vegetable, value: 'Carrot'),
+          ],
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [Item(type: ItemType.fruit, value: 'Apple')],
+          filter: ItemType.fruit,
+        ),
+      ],
+    );
+
+    blocTest<ListCubit<Item, ItemType>, Data<List<Item>, ItemType>>(
+      'is able to remove an item by value correctly',
+      build: () => cubit,
+      act: (cubit) async {
+        await cubit.load(ItemType.fruit);
+        cubit.add(Item(type: ItemType.vegetable, value: 'Carrot'));
+        cubit.remove(Item(type: ItemType.vegetable, value: 'Carrot'));
+      },
+      expect: () => [
+        Data<List<Item>, ItemType>.loading(
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [Item(type: ItemType.fruit, value: 'Apple')],
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [
+            Item(type: ItemType.fruit, value: 'Apple'),
+            Item(type: ItemType.vegetable, value: 'Carrot'),
+          ],
+          filter: ItemType.fruit,
+        ),
+        Data<List<Item>, ItemType>(
+          data: [Item(type: ItemType.fruit, value: 'Apple')],
+          filter: ItemType.fruit,
+        ),
+      ],
+    );
+
+    blocTest<ListCubit<Item, ItemType>, Data<List<Item>, ItemType>>(
```

Comment:
Let's add a test when we provide a wrong index and it throws error because that index will be out of range or negative